integration/tests/bft/signature-verification/signature_verification_test.go

```diff
@@ -0,0 +1,37 @@
+package signature_verification
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type SignatureVerificationTestSuite struct {
+	Suite
+}
+
+func TestSignatureVerification(t *testing.T) {
+	suite.Run(t, new(SignatureVerificationTestSuite))
+}
+
+// TestSignatureVerificationE2E ensures that the libp2p signature verification is working as expected.
+// The test network is set up with 3 corrupted AN, 2 attackers and one victim. 1 attacker AN that has message signing disabled, this
+// node will be used to send messages without signatures to the victim AN. These messages should never be
+// delivered to the victim as the signature verification is expected to fail. 1 attacker AN with message
+// signing enabled, this node will send messages that are expected to be delivered to the victim AN.
+func (s *SignatureVerificationTestSuite) TestSignatureVerificationE2E() {
```

Comment:
```suggestion
func (s *SignatureVerificationTestSuite) TestGossipSubSignatureRequirement() {
```
The test scenario is great, but I suggest a more specific naming for the test, as it may be deemed as a signature verification against corrupted signatures (i.e., impersonation), which it is not. Please also emphasize this in the documentation of the test as well as the package and file naming.

Reply:
https://github.com/onflow/flow-go/pull/3689/commits/08c35ffca7d3f65002a569e8eeeb0befd787453e